dialogflow/cloud-client/src/main/java/com/example/dialogflow/DetectIntentKnowledge.java

```diff
@@ -16,52 +16,50 @@
 
 package com.example.dialogflow;
 
-
 // Imports the Google Cloud client library
+
+import com.google.cloud.Tuple;
 import com.google.cloud.dialogflow.v2beta1.DetectIntentRequest;
 import com.google.cloud.dialogflow.v2beta1.DetectIntentResponse;
 import com.google.cloud.dialogflow.v2beta1.KnowledgeAnswers;
 import com.google.cloud.dialogflow.v2beta1.KnowledgeAnswers.Answer;
-import com.google.cloud.dialogflow.v2beta1.KnowledgeBaseName;
 import com.google.cloud.dialogflow.v2beta1.QueryInput;
 import com.google.cloud.dialogflow.v2beta1.QueryParameters;
 import com.google.cloud.dialogflow.v2beta1.QueryResult;
 import com.google.cloud.dialogflow.v2beta1.SessionName;
 import com.google.cloud.dialogflow.v2beta1.SessionsClient;
 import com.google.cloud.dialogflow.v2beta1.TextInput;
 import com.google.cloud.dialogflow.v2beta1.TextInput.Builder;
+import com.google.common.collect.Lists;
 
 import java.util.List;
-import java.util.UUID;
-import net.sourceforge.argparse4j.ArgumentParsers;
-import net.sourceforge.argparse4j.inf.ArgumentParser;
-import net.sourceforge.argparse4j.inf.ArgumentParserException;
-import net.sourceforge.argparse4j.inf.Namespace;
-
 
-/** DialogFlow API Detect Intent sample with querying knowledge connector. */
+/**
+ * DialogFlow API Detect Intent sample with querying knowledge connector.
+ */
 public class DetectIntentKnowledge {
-
   // [START dialogflow_detect_intent_knowledge]
+
   /**
    * Returns the result of detect intent with text as input.
    *
    * <p>Using the same `session_id` between requests allows continuation of the conversation.
    *
-   * @param projectId Project/Agent Id.
+   * @param projectId       Project/Agent Id.
    * @param knowledgeBaseId Knowledge base Id.
-   * @param sessionId Identifier of the DetectIntent session.
-   * @param languageCode Language code of the query.
-   * @param texts The texts to be processed.
+   * @param sessionId       Identifier of the DetectIntent session.
+   * @param languageCode    Language code of the query.
+   * @param texts           The texts to be processed.
+   * @return The KnowledgeAnswers found for each text.
    */
-  public static void detectIntentKnowledge(
+  public static List<Tuple<String, KnowledgeAnswers>> detectIntentKnowledge(
```

Comment:
Curious (correct me if I'm wrong): Is the `Tuple` more of a Python thing? That is specially added as part of google.cloud.java.

Could a better return that makes navigating the answers for all of a `text` be?
`Hashmap<String, List<KnowledgeAnswers>>`

Downside being that you have to instantiate the `List` for each `text`

Reply:
Originally I had this as a Map<String, KnowledgeAnswers>, but the downside is that maps aren't guaranteed order so you wouldn't actually know what the "last" response was. An ordered list of key value pairs was the best alternative I could think of. 

But as with the other conversation, that's more an issue of how this sample is set up in the first place: passing in multiple text inputs and then running them through a loop in the method instead of having the method just accept a single text and running it multiple times in the tests/usecase. Might be worth just refactoring it that way to simplify.

Reply:
True, but you know the order in which you sent the `texts` so you could use that to search for the answers for each `text`. 

Guess it's sort of a trade off, my only real concern is the use of the `Tuple` and how that flows with Java

Reply:
I can live with that